packages/compass-query-bar/src/modules/ai-query-request.ts

```diff
@@ -13,25 +15,69 @@ function getAIQueryEndpoint(): string {
   return process.env.DEV_AI_QUERY_ENDPOINT;
 }
 
+function getAIBasicAuth(): string {
+  if (!process.env.DEV_AI_USERNAME || !process.env.DEV_AI_PASSWORD) {
+    throw new Error(
+      'No AI auth information found. Please set the environment variable `DEV_AI_USERNAME` and `DEV_AI_PASSWORD`'
+    );
+  }
+
+  const authBuffer = Buffer.from(
+    `${process.env.DEV_AI_USERNAME}:${process.env.DEV_AI_PASSWORD}`
+  );
+  return `Basic ${authBuffer.toString('base64')}`;
+}
+
+const MAX_REQUEST_SIZE = 5000;
+const MIN_SAMPLE_DOCUMENTS = 1;
+
 export async function runFetchAIQuery({
   signal,
   userPrompt,
+  collectionName,
+  schema,
+  sampleDocuments,
 }: {
   signal: AbortSignal;
   userPrompt: string;
+  collectionName: string;
+  schema?: SimplifiedSchema;
+  sampleDocuments?: Document[];
 }) {
+  let msgBody = JSON.stringify({
+    userPrompt,
+    collectionName,
+    schema,
+    sampleDocuments,
+  });
+  if (msgBody.length > MAX_REQUEST_SIZE) {
+    // When the message body is over the max size, we try
+    // to see if with fewer sample documents we can still perform the request.
+    // If that fails we throw an error indicating this collection's
+    // documents are too large to send to the ai.
+    msgBody = JSON.stringify({
+      userPrompt,
+      collectionName,
+      schema,
+      sampleDocuments: sampleDocuments?.slice(0, MIN_SAMPLE_DOCUMENTS),
```

Comment:
Do we want to iteratively reduce the # of sampleDocuments instead of always going from the full list to a single item? Or do we think that it probably doesn’t matter?

Reply:
I think it probably doesn't matter, at least for now. As we start to learn more about how we can best prompt the ai we can update it if we feel it would help.